api/v1_projects.inc

```diff
@@ -184,6 +192,28 @@ function create_or_update_project($project)
         }
     }
 
+    // enforce specific field authorization for non-SAs
+    if (!user_is_a_sitemanager()) {
+        // project creation
+        if (!$orig_project->projectid) {
+            if (User::current_username() != $project->username) {
+                throw new UnauthorizedError("You do not have permission to create projects for another PM");
+            }
+            if (User::current_username() != 'BEGIN' && $project->difficulty == "beginner") {
```

Comment:
There is a user with the name "BEGIN" that is the only person who can set difficulty to beginner? I never knew this.

Reply:
Actually, siteadmins should also be able to set the difficulty to beginner.

Reply:
They can. These inner checks only apply to non-SAs.

Reply:
> There is a user with the name "BEGIN" that is the only person who can set difficulty to beginner?

Yeah, this is one of several pgdp.net-specific things that is kind baked into our code and hard to extract.